libbeat/common/kubernetes/util.go

```diff
@@ -30,10 +30,29 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 
+	"github.com/pkg/errors"
+
 	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
-const defaultNode = "localhost"
+const namespaceFilePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+type HostDiscoveryUtils interface {
+	GetNamespace() (string, error)
+	GetPodName() (string, error)
+	GetMachineID() string
+}
+
+// DiscoverKubernetesNodeOpts includes options for discovering kubernetes node
```

Comment:
Nit. This includes more than options :slightly_smiling_face: it also includes a fully fleshed Kubernetes client and an implementation of discovery utils.

Maybe this is a signal that `DiscoveryKubernetesNode()` could be a member of an object containing these fields, and maybe receive the options as parameter.

Reply:
I changed the name to DiscoverKubernetesNodeParams which is more clear to me